doc/source/user_guide/text.rst

```diff
@@ -264,25 +264,30 @@ i.e., from the end of the string to the beginning of the string:
        ["A", "B", "C", "Aaba", "Baca", "", np.nan, "CABA", "dog", "cat"], dtype="string"
    )
    s3
-   s3.str.replace("^.a|dog", "XX-XX ", case=False)
+   s3.str.replace("^.a|dog", "XX-XX ", case=False, regex=True)
 
-Some caution must be taken to keep regular expressions in mind! For example, the
-following code will cause trouble because of the regular expression meaning of
-``$``:
+Some caution must be taken when dealing with regular expressions! The current behavior
+is to treat single character patterns as literal strings, even when ``regex`` is set
+to ``True``. (This behavior is deprecated and will be removed in a future version so
+that the ``regex`` keyword is always respected.) For example, the following code will
+cause trouble because of the regular expression meaning of ``$``:
 
 .. ipython:: python
 
    # Consider the following badly formatted financial data
    dollars = pd.Series(["12", "-$10", "$10,000"], dtype="string")
 
-   # This does what you'd naively expect:
-   dollars.str.replace("$", "")
+   # Here $ is treated as a literal character
+   dollars.str.replace("$", "", regex=True)
```

Comment:
@dsaxton I don't think we should show (and thus encourage) the use of `regex=True` with a single character, as it will not be treated as a regex regardless of the keyword .. 

(something I am wondering: what would that actually do if it was interpreted as regex? So assume we in the future start honoring the `regex=True`, how would that change behaviour?)

Reply:
I don't think we're necessarily encouraging this behavior because in the paragraph above we're explicitly warning about it and saying the behavior is deprecated. Does it make sense perhaps to not even give an example? 

I think the "correct" behavior is for this to be a no-op (we're replacing the already empty "last position" with an empty string, which does nothing):
```python
[ins] In [3]: re.sub("$", "abc", "")
Out[3]: 'abc'
```

Reply:
this also needs a version changed; i also agree that we should not be encouraging this example

Reply:
Removed / versionchanged added